store.go

```diff
@@ -1545,6 +1547,9 @@ func (s *store) putLayer(rlstore rwLayerStore, rlstores []roLayerStore, id, pare
 			GIDMap:         copySlicePreferringNil(gidMap),
 		}
 	}
+	if !writeable && s.imageStoreDir != "" {
+		options.ImageStore = true
+	}
```

Comment:
I don’t generally see why this logic should exist at this layer of the call stack. `layerStore` was already initialized with the imageStore location, it seems to me that it can make this decision on its own.

Alternatively… for `imageStore`, the search, and the decision where to write is managed at the store level. There is no `multipleLockFile` for images. This looks _so_ different, and I don’t understand the design nearly enough to tell whether the difference is necessary.

Reply:
the difference is necessary because images are always directed to the image store.  Instead for layers, we write only the RO layers to the imagestore; container layers are written to the main store.

Reply:
should remapped images be always created on the image store?

I think it is better if they are created on the same store where their parent layer is.  I've added a patch to the PR to implement this behavior

Reply:
Fine, I can see how that is a good location to choose

… but now, again, we have one part of the code that decides the location in `layers.go` and another here in `store.go`.
WDYT about `layerStore` explicitly tracking whether imageDir is set (compare https://github.com/containers/storage/pull/2258#discussion_r1960040690 ), and then this condition can be moved inside the new `pickLocation`.

Reply:
> and then this condition can be moved

no, that doesn’t work _that_ easily because, AFAICS, we can’t currently tell between “ordinary” RO layers and the ID-mapping layers from `imageTopLayerForMapping`; we would still need to add a new option to `create`. Still, an `thisIsAnIDMappingLayer` option set in `store.Go`, and all location decisions centralized in `pickLocations`, seems worth considering.